### PR TITLE
release: v1.0.0a10 

### PR DESCRIPTION
**Pull request by lnielsen**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -72,9 +72,9 @@
     'Flask-BabelEx>=0.9.3',
     'mistune>=0.7.2',
     'cchardet>=1.0.0',
-    'invenio-assets>=1.0.0a3',
-    'invenio-pidstore>=1.0.0a6',
-    'invenio-records-ui>=1.0.0a7',
+    'invenio-assets>=1.0.0b4',
+    'invenio-pidstore>=1.0.0b1',
+    'invenio-records-ui>=1.0.0a8',
```
Comment (jirikuncar) [inline setup.py]: shouldn't the version bump be part of previous commit?

Comment (lnielsen) [inline setup.py]: Yes, but I will merge anyway as we unfortunately had a production deployment so all previews are broken currently on zenodo. Mistake on our side but need fast track not this one.

**Comment by lnielsen** [comment]
LGTM

**Comment by lnielsen** [comment]
@jirikuncar Too late, tag was already pushed.

**Comment by tiborsimko** [comment]
> Too late, tag was already pushed.

It would have been easier to amend things (and achieve "proper" commit atomicity) if the release commit and tagging were separated from the bug fix. We should probably revive past discussions on (i) separating PRs for bug fixes and features from PRs for releases, and on (ii) pinning also github-dependencies, not only pypi-dependencies, for ultra fast deployment of hot fixes on services. (Because nothing can be faster than those?!)

There are pros and cons, we started to discuss about these before, e.g. see my musings [looking out for some future stable model here](https://github.com/inveniosoftware/invenio-github/pull/30#issuecomment-248859188), but we did not conclude at the time... Let's try to revive this again among maintainers?

**Comment by lnielsen** [comment]
Listen, let's try to be a bit practical here. Here's what happened:

- inveniosoftware/invenio-records-ui#63 changed API and this change wasn't advertised in the PR, and I didn't detect it when I reviewed and merged the PR.
- We've had zenodo/zenodo#881 hanging around for quite a while which are fixing various user facing problems, and have been blocked on it due to various other Invenio PRs on deposit, accounts, records-rest, files-rest, records-files. Finally last blocker inveniosoftware/invenio-deposit#143 was merged and released and we could deploy today. 
- We deployed on first sandbox then production, but too late detected that *all* previews were broken due to the API change in Records-UI.
- We had scheduled a strategy meeting for the entire day, and thus urgently need to fix the issue. We could have fixed the it directly in Zenodo with a wrapper function, instead we choose for the benefit of everyone else to fix Invenio-Preview and get a fast release.
- The tag was already pushed (but master not due to missing LGTM) when @jirikuncar  made his comment and we were already in the meeting, hence short message and no time to fix.

So, it's an unfortunate chain of events that led to this, but let's try to be a bit practical. We're talking about a couple of version bumps in a release commit. We got a problem fixed for the benefit of all of Invenio, when we could just have done it by ourselves and let the rest of you waste the time in fixing it. I think you are making it really hard to be motivated about contributing to Invenio if tiny mistakes like this are blown out of proportion when we in fact were thinking about the Invenio community and being unselfish. 

So, yes, release commits should not include other stuff but it's a tiny mistake. On the other hand I think pushing away Invenio contributors is far my harmful and with long-term devastating effect to Invenio than a couple of version bumps in a release commit.

**Comment by tiborsimko** [comment]
I wasn't trying to blow anything out of proportion... at least it wasn't my intention. Please read my [previous disclaimers](https://github.com/inveniosoftware/invenio-github/pull/30#issuecomment-248859188) that I know fully well apply here as well 😄  

I was simply questioning the need to quick-push a release in order to quick-fix a problem on a service. I'm afraid that this approach is not scalable in a world when a service depends on Np packages that are contributed to by Nd developers and where releases are done by Nm maintainers. Because, if a service needs an urgent bug fix for a package that some other service maintains, then one cannot realistically assume anything about the expected timeline for when a patch might be reviewed and accepted, and for when a new release might be pushed to PyPI. Precisely because one depends on the community. *(Which I know is not the case here, see above disclaimers.)* 

Now that more and more packages are coming out of alpha, and that more and more package maintainers enter the game, hasn't the time come to muse about some more future-proof deployment model? For example, pinning github-based dependencies in service deployments, good or bad? Just an example of the musings I was having in mind.

**Comment by ludmilamarian** [comment]
Regarding quick-pushing a release in order to fix a problem on a service (but potentially on many other services), I would think it should be left at the decision of the maintainer of the module (BTW, this discussion touches a bit the entire community as users/services would base their deployment strategy on this, so probably a dedicated RFC would be better, as many people are probably  watching neither `invenio-previewer` nor `invenio-github`). The maintainer will know if the fix is worth a release or not, and in this particular case the maintainer decided it's worth a release, so it should be ok.
Related to the deployment model, I would prefer for CDS to spend zero time on production deployment (which today I must admit is quite time consuming due to our particular setup) and invest time in testing and automating the deployment which in my mind would be continuous deployment using unpinned Invenio dependencies.

**Pull request merged**